src/color-gamut/color-gamut.js

```diff
@@ -10,162 +11,111 @@ const Self = class ColorGamut extends HTMLElement {
-			this.#setGamut();
-			return;
-		}
-
-		let gamut = this.gamuts.find(gamut => gamut.id === "none" || this.#color.inGamut(gamut.id));
-		this.#setGamut(gamut);
-	}
-
-	#setGamut (gamut) {
-		let oldGamut = this.#gamut;
-
-		if (gamut === oldGamut) {
-			return;
+	propChangedCallback ({name, prop, detail: change}) {
+		if (name === "gamuts") {
+			this.style.setProperty("--gamut-count", this.gamuts.length - 1);
 		}
 
-		this.#gamut = gamut;
-		this.#label.textContent = this.#gamut?.label ?? "";
-
-		if (!gamut) {
-			this.removeAttribute("gamut");
-			this.removeAttribute("level");
-			this.style.removeProperty("--gamut-level");
-
-			return;
-		}
-
-		this.setAttribute("gamut", gamut.id);
-		this.setAttribute("level", gamut.level);
-		this.style.setProperty("--gamut-level", gamut.level);
-
-		this.dispatchEvent(new CustomEvent("gamutchange", {
-			detail: {oldGamut, gamut},
-			bubbles: true,
-		}));
-	}
-
-	#color;
-	get color () {
-		return this.#color;
-	}
-
-	set color (color) {
-		this.#color = Color.get(color);
-		this.#render();
-	}
-
-	static observedAttributes = ["gamuts"];
-
-	attributeChangedCallback (name, oldValue, newValue) {
-		if (!name && this.hasAttribute("gamuts") || name === "gamuts") {
-			newValue ??= this.getAttribute("gamuts");
-
-			if (oldValue !== newValue) {
-				this.gamuts = this.constructor.parseGamuts(newValue ?? this.constructor.defaultGamuts);
-				this.#render();
+		if (name === "gamut") {
+			if (this.gamutInfo) {
+				this.style.setProperty("--gamut-level", this.gamutInfo.level);
+				this.style.setProperty("--gamut-label", this.gamutInfo.label);
+				this.style.setProperty("--gamut-id", `"${ this.gamutInfo.id }"`);
+			}
+			else {
+				this.style.removeProperty("--gamut-level");
+				this.style.removeProperty("--gamut-label");
+				this.style.removeProperty("--gamut-id");
 			}
 		}
 	}
 
-	static defaultGamuts = ["srgb", "p3", "rec2020", "prophoto"];
-
-	static parseGamuts (gamuts) {
-		if (!gamuts) {
-			return [];
-		}
-
-		if (typeof gamuts === "string") {
-			gamuts = gamuts.trim().split(/\s*,\s*/);
-		}
-		else if (!Array.isArray(gamuts) && typeof gamuts === "object") {
-			// Object
-			return Object.entries(gamuts).map(([id, label]) => {id, label});
-		}
-
-		let ret = gamuts.map((gamut, level) => {
-			if (gamut?.id && "label" in gamut) {
-				// Already in the correct format
-				return gamut;
+	static props = {
+		color: {
+			type: Color,
+			parse (color) {
+				try {
+					return Color.get(color);
+				}
+				catch (e) {
+					console.warn("<color-gamut>:", e);
+					return null;
+				}
 			}
+		},
+		gamuts: {
+			type: Array,
+			default: "srgb, p3, rec2020, prophoto",
+			parse (gamuts) {
+				if (!gamuts) {
+					return [];
+				}
+
+				if (typeof gamuts === "string") {
+					gamuts = gamuts.trim().split(/\s*,\s*/);
+				}
+				else if (!Array.isArray(gamuts) && typeof gamuts === "object") {
+					// Object
+					return Object.entries(gamuts).map(([id, label]) => {id, label});
```

Comment:
```suggestion
					return Object.entries(gamuts).map(([id, label]) => ({id, label}));
```

Reply:
Do I get it correctly that we expect to get an array of objects?

Reply:
Oh yes! But this is not new code, how did it work before? So weird.